# 2022_Papilio_HybridIncompatibilityMapping

Description of source data and scripts for all figures

==== MAIN FIGURES ====

Fig. 1

 - Panel A
     * Schematic figure, no source data are provided

 - Panel B
     * Schematic figure, no source data are provided

 - Panel C
     * Source data folder(s):
         SourceData/Fig1/Fig1C
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 5

 - Panel D
     * Source data folder(s):
         SourceData/Fig1/Fig1D
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 5

 - Panel E
     * Source data folder(s):
         SourceData/Fig1/Fig1E
     * Source code:
         SourceData/Code.JupyterLab/Code01_SequencingData.ipynb: Section 2

 - Panel F
     * Source data folder(s):
         SourceData/Fig1/Fig1F
     * Source code:
         SourceData/Code.JupyterLab/Code01_SequencingData.ipynb: Section 2

Fig. 2

 - Panels A-G
     * Source data folder(s):        
         SourceData/Fig2+S1toS2    
     * The "Raw" folder contains unedited images.
     * Two edited images used in Fig2 is also included for each subfigure.

 - Panels H-L
     * Source data folder(s): 
        SourceData/Fig2+S1toS2      
     * The "Raw" folder (unzipped) contains unedited confocal data in .czi format.
     * Edited images are included with both monochrome and merged versions.

Fig. 3

 - Panel A
     * Source data folder(s):
         SourceData/Fig3+S8toS10/Fig3A_3B_3C_3D_3E
     * Source code:
         D(DB): SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.1
         B(BD): SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 2.3

 - Panel B
     * Source data folder(s): 
         SourceData/Fig3+S8toS10/Fig3A_3B_3C_3D_3E
     * Source code:
         D(DB): SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.1
         B(BD): SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.2

 - Panel C
     * Source data folder(s): 
         SourceData/Fig3+S8toS10/Fig3A_3B_3C_3D_3E
     * Source code:
         D(DB): SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.1
         B(BD): SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.2

 - Panel D
     * Source data folder(s): 
         SourceData/Fig3+S8toS10/Fig3A_3B_3C_3D_3E
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Run all of Sections 4.1 and 4.2

 - Panel E
     * Source data folder(s): 
         SourceData/Fig3+S8toS10/Fig3A_3B_3C_3D_3E
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Run all of Sections 4.1 and 4.2

Fig. 4

 - Note 1: For Heliconius analysis, all data are from SourceData/Fig4-Heliconius+S11C/dat.4.qtl.lumped.csv. This file contains Heliconius ovary dysgenesis data from https://doi.org/10.1111/mec.16272

 - Panel A (Heliconius)
     * Source data folder(s): 
         SourceData/Fig4-Heliconius+S11C
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 6.1
 
 - Panel A (Papilio) 
     * Source data folder(s): 
         SourceData/Fig4-Papilio+S13
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.2

 - Panel B (Heliconius)
     * Source data folder(s): 
         SourceData/Fig4-Heliconius+S11C
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 6.1
 
 - Panel B (Papilio)
     * Source data folder(s): 
         SourceData/Fig4-Papilio+S13
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.2

 - Panel C (Heliconius) 
     * Source data folder(s): 
         SourceData/Fig4-Heliconius+S11C
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 6.2
 
 - Panel C (Papilio) 
     * Source data folder(s):
         SourceData/Fig4-Papilio+S13
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.3

 - Panel D
     * Schematic figure, no source data are provided
     
 - Panel E (Heliconius) 
     * Source data folder(s): 
         SourceData/Fig4-Heliconius+S11C
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 6.3
 
 - Panel E (Papilio) 
     * Source data folder(s): 
         SourceData/Fig4-Papilio+S13
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.3
 
 - Panel F (Heliconius)
     * Source data folder(s): 
         SourceData/Fig4-Heliconius+S11C
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 6.3
 
 - Panel F (Papilio) 
     * Source data folder(s): 
         SourceData/Fig4-Papilio+S13
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.3
 
 
 
==== SUPPLEMENTARY FIGURES ====

Fig. S1-S2

 * Source data folder(s): 
        SourceData/Fig2+S1toS2      
 * The "Raw" folder (unzipped) contains unedited confocal data in .czi format.
 * Edited images are included with both monochrome and merged versions.

Fig. S3

  * Source data folder(s): 
         SourceData/FigS3
  * Source code:
         SourceData/Code.JupyterLab/Code01_SequencingData.ipynb: Section 1
  * Note: Source data file 04.0_IBD.NgsRelate.zip contains results from the NGSRelate software.
         

Fig. S4

  * Source data folder(s): 
         SourceData/FigS4
  * Source code:
         SourceData/Code.JupyterLab/Code01_SequencingData.ipynb: Section 2
  * Note 1: Source data file CorrectedReferenceGenome.zip is the corrected reference genome used for all analyses. It is in .fasta format.
  * Note 2: Source data file DenovoMarkerOrder_on_CorrectedRefGenome.zip contains all outputs from the LepMap3/OrderMarkers2 module that uses genotype likelihoods and pedigree information to generate a new marker order. Use script "OrderMarkers2_ReOrder.sh" from the script repo.
         

Fig. S5-S7

  * Source data folder(s): 
         SourceData/FigS5toS7
  * Source code:
         SourceData/Code.JupyterLab/Code01_SequencingData.ipynb: Section 2
  * Note: The source data file PedigreeAncestryInGrandparentalPhase.zip contains all outputs from the LepMap3/OrderMarkers2 module that uses genotype likelihoods and pedigree information to impute ancestry at each marker. Ancestry is phased according to the sex of grandparents. Use script "OrderMarkers2.sh" from the GitHub repo.

Fig. S8

 - Panel A     
     * Source data folder(s): 
         SourceData/Fig3+S8toS10
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.1

 - Panel B     
     * Source data folder(s): 
         SourceData/Fig3+S8toS10
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 2.3

 - Panel C
     * Source data folder(s): 
         SourceData/Fig3+S8toS10
     * Source code:
         See previous two panels

Fig. S9

 - Panel A
     * Source data folder(s): 
         SourceData/Fig3+S8toS10
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.1

 - Panel B
     * Source data folder(s): 
         SourceData/Fig3+S8toS10
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.2

Fig. S10

 - Panel A
     * Source data folder(s):
         SourceData/Fig3+S8toS10
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.1

 - Panel B
     * Source data folder(s): 
         SourceData/Fig3+S8toS10
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 4.2

Fig. S11

 - Panel A
     * Source data folder(s): 
         SourceData/FigS11AB
     * Source code:
         Data in SpeciesAncestry.B0D1.zip can be directly visualized to get the figure

 - Panel B
     * Source data folder(s): 
         SourceData/FigS11AB
     * Source code:
         Data in SpeciesAncestry.B0D1.zip can be directly visualized to get the figure
     * Note: This file contains ancestry at each marker phased according to species (bianor=0, dehaanii=1). These data are directly transformed from files in SourceData/FigS5toS7/PedigreeAncestryInGrandparentalPhase.zip.

 - Panel C
     * Source data folder(s): 
         SourceData/Fig4-Heliconius+S11C
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 6.4

Fig. S12

  * Source data folder(s): 
         No source data are needed
  * Source code:
         SourceData/Code03_PolygenicGhostQTL.ipynb

Fig. S13

 - Panel A
     * Source data folder(s): 
         SourceData/Fig4-Papilio+S13
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.3

 - Panel B
     * Source data folder(s): 
         SourceData/Fig4-Papilio+S13
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 3.2

Fig. S14

  * Source data folder(s): 
         No source data are needed
  * Source code:
         SourceData/Code03_PolygenicGhostQTL.ipynb

Fig. S15

 - Panel A
     * Source data folder(s): 
         SourceData/FigS15/FigS15A
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 2.1

 - Panel B
     * Source data folder(s): 
         SourceData/FigS15/FigS15B
     * Source code:
         SourceData/Code.JupyterLab/Code02_Mapping.ipynb: Section 2.2

Fig. S16

  * Source data folder(s): 
         SourceData/FigS16
  * Source code:
         SourceData/Code.JupyterLab/Code01_SequencingData.ipynb: Section 3
         

==== OTHER SOURCE DATA & SUMMARY OF SOURCE CODE FOLDERS====

LepMap3-SourceData

  * LepMap3_SourceData-Family_Info_Finalized_withPseudoGrandParents_transposed.txt.zip
         
      This file is the pedigree file ready-to-use in LepMap3. Note that it contains pseudo grandparents for families missing grandparents in sequencing. Pseudo grandparents are simply created from fixed SNPs in all existing grandparents and adding them to the original vcf files containing genotype likelihoods.

  * LepMap3_SourceData-vcf_files.zip
         
      The vcf files containing genotype likelihoods for LepMap3 to use. Note that it contains the aforementioned pseudo grandparents.


Code.NGSRelate

  * Code used for inferring kinship from low-coverage sequencing data

Code.LepMap3
    
  * Code used for all LepMap3 analysis

Code.JupyterLab

  * Code used for all Julia and R analysis in .ipynb format





